lib/graphql/schema_comparator/diff/schema.rb

```diff
@@ -34,6 +34,22 @@ def diff
           changes
         end
 
+        def changes_in_added_type(type)
+          changes = [Changes::TypeAdded.new(type)]
+          if type.graphql_definition.is_a?(GraphQL::ObjectType)
+            type.interfaces.each do |interface|
+              if old_types[interface.graphql_name]
+                changes << Changes::ObjectTypeInterfaceAdded.new(interface, type)
```

Comment:
I guess ideally this would only be dangerous for existing types right? It's good to capture this change for new types, but it's not actually dangerous at this point.

Reply:
Exactly. It's only dangerous when implementing an existing interface.